compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10533,16 +10531,31 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
       TR::Register *baseReg1Temp = NULL;
       TR::Register *baseReg2Temp = NULL;
 
+
+      /*
+         If (source1.base + source1.displacement) is equal to (source2.base + source2.displacement) then source1 and
+      source2 point to the same location and the arrays are equal. In this case, we can short circuit the compare
+      instructions and return "equal" result to improve the performance. There are two possible scenarios when comparing
+      addresses for shourt circuiting: 1) If source1.displacement is equal to source2.displacement, then we just neet to
```

Comment:
I would go with something like this, 

```
It may be possible that displacement for source1 / source2 operand is very large for CLC or displacement for both source1 and source2 operand are different. In such case, both source address will be forced into register to allow generating a quick test to compare if source1 and source2 array addresses are equal.
```

Reply:
Thanks for your suggestion. I updated the comment as you suggested.